packages/babel-helper-create-class-features-plugin/src/misc.ts

```diff
@@ -142,21 +142,37 @@ export function extractComputedKeys(
     // Make sure computed property names are only evaluated once (upon class definition)
     // and in the right order in combination with static properties
     if (!computedKey.isConstantExpression()) {
-      const ident = path.scope.generateUidIdentifierBasedOnNode(
-        computedNode.key,
-      );
-      // Declaring in the same block scope
-      // Ref: https://github.com/babel/babel/pull/10029/files#diff-fbbdd83e7a9c998721c1484529c2ce92
-      path.scope.push({
-        id: ident,
-        kind: "let",
-      });
-      declarations.push(
-        t.expressionStatement(
-          t.assignmentExpression("=", t.cloneNode(ident), computedNode.key),
-        ),
-      );
-      computedNode.key = t.cloneNode(ident);
+      const scope = path.scope;
+      const isUidReference =
+        t.isIdentifier(computedKey.node) && scope.hasUid(computedKey.node.name);
+      const isMemoiseAssignment =
+        computedKey.isAssignmentExpression({ operator: "=" }) &&
+        t.isIdentifier(computedKey.node.left) &&
+        scope.hasUid(computedKey.node.left.name);
```

Comment:
Sorry, I didn't understand why `hasUid` is used, can you explain it? Thanks!

Reply:
Good question. The `hasUid` checks whether the identifier is generated from `scope.generateUidIdentifier`, if it is, it is very likely generated by official Babel plugins, intended to be a memoiser. If not, it is from user code and we should handle it more conservatively.